src/interpreter/utils.ts

```diff
@@ -293,6 +294,22 @@ export const mergeVariables = (
   return result;
 };
 
+export const getProfileUsecases = (
+  profile: ProfileDocumentNode
+): { name: string; safety?: 'safe' | 'unsafe' | 'idempotent' }[] => {
+  const usecases: {
+    name: string;
+    safety?: 'safe' | 'unsafe' | 'idempotent';
+  }[] = [];
+  profile.definitions.forEach(d => {
```

Comment:
Yep, type inference is borked that way 😩 property in map is not UseCaseDefinition node 🤨 I can cast it but imho casting is worse then for each 🤷

Reply:
![image](https://user-images.githubusercontent.com/13323507/125437939-c3b8d32e-08bc-4ba7-95e3-a313ce88aed1.png)

Reply:
also, no single letter variables pls

Reply:
Fixed, weird is that this infers types `.filter(isUseCaseDefinitionNode)` and this doesn't `.filter(definition => isUseCaseDefinitionNode(definition))`

Reply:
not weird at all. in `.filter(isUseCaseDefinitionNode)` the type of the callback is `(x) => x is UseCaseDefinitionNode`, but in the other it is `() => boolean`, because you're making a new function without explicit type

Reply:
Thanks